File: models/bom.py
from sqlalchemy import Column, ForeignKey, Numeric, String

from database import Base


class Bom(Base):
    __tablename__ = "bom"

    id = Column(String, primary_key=True)
    jewelry_id = Column(String, ForeignKey("jewelry.id"), nullable=False)
    part_id = Column(String, ForeignKey("part.id"), nullable=False)
    qty_per_unit = Column(Numeric(10, 4), nullable=False)

File: frontend/src/api/bom.js
import api from './index'

export const getBom = (jewelryId) => api.get(`/bom/${jewelryId}`)
export const setBom = (jewelryId, partId, qtyPerUnit) =>
  api.put(`/bom/${jewelryId}/${partId}`, { qty_per_unit: qtyPerUnit })
export const deleteBom = (jewelryId, partId) =>
  api.delete(`/bom/${jewelryId}/${partId}`)
